### PR TITLE
Fix ManyMouse support in Windows builds

### PR DESCRIPTION
**Pull request by FeralChild64**
# Description

- modified Windows-specific `config.h` file to restore _ManyMouse_ compilation on Windows (regression)
- on Windows prevent _ManyMouse_ usage together with raw mouse input; these are not compatible

This definitely has to be a part of 0.82.1 release.

## Related issues

Fixes https://github.com/dosbox-staging/dosbox-staging/issues/4026

# Release notes

Fixed dual-mouse gaming support not working on Windows.

# Manual testing

- try using `mousectl -all` and `mousectl DOS -map` commands under Windows - they should be blocked if `mouse_raw_input` is set to `true`
- `config -set mouse_raw_input true` command should not disrupt mouse mapping (check on Windows, in windowed mode)

# Checklist

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [x] provided the release notes draft (for significant user-facing changes).
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Review by weirddan455 — APPROVED** [review]
Looks good to me.

File: src/hardware/input/mouse_interfaces.h [hunk 1]
```diff
@@ -33,6 +33,7 @@ void MOUSE_NotifyDisconnect(const MouseInterfaceId interface_id);
 
 void MOUSE_UpdateGFX();
 bool MOUSE_IsCaptured();
+bool MOUSE_IsInputRaw();
```
Comment (johnnovak) [inline src/hardware/input/mouse_interfaces.h:36]: MOUSE_IsRawInput would be better

**Review by johnnovak — APPROVED** [review]

File: src/hardware/input/mouse_interfaces.h (hunk 1 above)
Comment (FeralChild64) [reply src/hardware/input/mouse_interfaces.h]: Changed, also variable names.

**Comment by FeralChild64** [comment]
@johnnovak Ok, I have changed the release notes.

**Comment by johnnovak** [comment]
@FeralChild64 Updated the release notes draft to mention The Settlers. Are you aware any other games that were affected by this regressions?

As a general note, we should put our "user hat" on when writing the release notes. They have zero care about the "how" on a technical level, they only care about "how does this affect me" and that's what we should focus on.

**Comment by johnnovak** [comment]
@Python-Exoproject Dual-mouse gaming in The Settlers is affected by this regression. So pretty niche, yeah. Will be fixed in 0.82.1

**Comment by FeralChild64** [comment]
@johnnovak I know only about Settlers and Settlers II.

I was unable to find anything else and asking on various forums only wasted my time - I got a lot of answers, all happened to be wrong ("ah, indeed, it supports a mouse and a joystick at the same time, not two mice, sorry").

**Comment by johnnovak** [comment]
> @johnnovak I know only about Settlers and Settlers II.
> 
> I was unable to find anything else and asking on various forums only wasted my time - I got a lot of answers, all happened to be wrong ("ah, indeed, it supports a mouse and a joystick at the same time, not two mice, sorry").

Thanks, I'll mention Settlers II too 👍🏻

**Comment by johnnovak** [comment]
> [Dual mouse](https://ofp-faguss.com/blog/2nd-mouse-com-port-pc/) support in Serious Sam (First encounter, Second encounter), but that's Win9x.

Interesting. We can't really mention that as we explicitly say "no Win 9x" support on our about page, though 😆

**Pull request merged**